### PR TITLE
chore: add Gemini newest experimental models (close #7121)

### LINKED ISSUE #7121: Add Gemini Pro Experimental
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing? Tell me about your story.

Google [added the experimental Gemini 1.5 Pro](https://x.com/OfficialLoganK/status/1819041453978357899) last week.

It's an open question to add experimental models to Dify, but I think we should do it this time for the following reasons;
- The model got 1st place in Chatbot Arena
- The users can't add new Gemini models to Dify manually

I already prepared the yaml file, but I open this issue because I saw that I should do so in the contribution guide and I think I should hear comments from others.

### 2. Additional context or comments

_No response_

### 3. Can you help us with this feature?

- [X] I am interested in contributing to this feature.

### PR DESCRIPTION
**Pull request by CXwudi**
# Checklist:

> [!IMPORTANT]  
> Please review the checklist below before submitting your pull request.

- [x] Please open an issue before creating a PR or link to an existing issue
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas (no code changes)
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods (no code changes)

# Description

Close #7121, added new experimental models from Gemini according to [documentation](https://ai.google.dev/gemini-api/docs/models/gemini). Specifically, the following models are added:

1. `gemini-1.5-pro-exp-0827`
2. `gemini-1.5-pro-exp-0801`
3. `gemini-1.5-flash-exp-0827`
4. `gemini-1.5-flash-8b-exp-0827`

Also, the max context size for `gemini-1.5-pro-latest` has been updated to 2M.

This PR doesn't have any code changes, but dependencies are updated to access the newest models

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [x] Improvement, including but not limited to code refactoring, performance optimization, and UI/UX improvement
- [x] Dependency upgrade

# Testing Instructions

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Setup the local environmental via devcontainer and try running some LLM query in Chatbot, Agent and Workflow mode

Untitled picture
Screenshot 2024-09-21 163106
Screenshot 2024-09-21 163649
Screenshot 2024-09-21 165809

### CODE REVIEW

File: web/yarn.lock
Comment (crazywoola) [inline web/yarn.lock:1]: This seems unrelated to this PR

Comment (CXwudi) [reply web/yarn.lock:1]: Sorry, let me revert it when I have time, should be soon

Comment (CXwudi) [reply web/yarn.lock:1]: Done @crazywoola

**Review by crazywoola — DISMISSED** [review]

**Comment by crazywoola** [comment]
It seems you need to sync the deps again.

**Comment by CXwudi** [comment]
@crazywoola Yeap, I just did

**Review by crazywoola — APPROVED** [review]
